src/app/components/Tokens/TokenTransfers.tsx

```diff
@@ -0,0 +1,217 @@
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  width: iconSize,
+  height: iconSize,
+  color: COLORS.eucalyptus,
+  backgroundColor: COLORS.lightGreen,
+  borderRadius: iconSize,
+  marginLeft: theme.spacing(3),
+  marginRight: `-${theme.spacing(4)}`,
+}))
+
+type TableRuntimeEvent = RuntimeEvent & {
+  markAsNew?: boolean
+}
+
+/**
+ * This is a wrapper around RoundedBalance which is able to find out the ticker by looking at an event
+ * @constructor
+ */
+const DelayedEventBalance: FC<
+  Omit<Exclude<Parameters<typeof RoundedBalance>[0], undefined>, 'ticker'> & {
+    event: RuntimeEvent
+  }
+> = props => {
+  const { t } = useTranslation()
+  const [oasisAddress, setOasisAddress] = useState<string | undefined>(undefined)
+  const { event, value, ...rest } = props
+
+  if (event.layer === Layer.consensus) {
+    throw AppErrors.UnsupportedLayer
+  }
+
+  const b64Address = event.body.address
+  const ethAddress = b64Address ? getEthAccountAddressFromBase64(b64Address) : undefined
+  useEffect(() => {
+    if (ethAddress) {
+      getEvmBech32Address(ethAddress).then(setOasisAddress)
+    }
+  }, [ethAddress])
+
+  const tokenQuery = useGetRuntimeEvmTokensAddress(event.network, event.layer, oasisAddress || '0x0')
+
+  const token = tokenQuery.data?.data
+  const realValue = value === undefined ? undefined : fromBaseUnits(value, token?.decimals || 0)
+  const ticker = token?.symbol || t('common.missing')
+
+  return <RoundedBalance {...rest} value={realValue} ticker={ticker} />
+}
+
+type TokenTransfersProps = {
+  transfers?: TableRuntimeEvent[]
+  ownAddress?: string
+  isLoading: boolean
+  limit: number
+  pagination: false | TablePaginationProps
+  verbose?: boolean
+}
+
+export const TokenTransfers: FC<TokenTransfersProps> = ({
+  isLoading,
+  limit,
+  pagination,
+  transfers,
+  ownAddress,
+  verbose,
+}) => {
+  const { t } = useTranslation()
+  const tableColumns: TableColProps[] = [
+    { key: 'hash', content: t('common.hash') },
+    { key: 'block', content: t('common.block') },
+    { key: 'type', content: t('common.type'), align: TableCellAlign.Center },
+    ...(verbose
+      ? [
+          { key: 'from', content: t('common.from'), width: '150px' },
+          { key: 'to', content: t('common.to'), width: '150px' },
+          { key: 'value', align: TableCellAlign.Right, content: t('common.value'), width: '250px' },
+        ]
+      : []),
+  ]
+  const tableRows = transfers?.map(transfer => {
+    const fromAddress = transfer.evm_log_params?.find(param => param.name === 'from')?.value as
+      | string
+      | undefined
+    const toAddress = transfer.evm_log_params?.find(param => param.name === 'to')?.value as string | undefined
+    const value = transfer.evm_log_params?.find(param => param.name === 'value')?.value as string | undefined
+    const isMinting = fromAddress === NULL_ADDRESS
+    const isBurning = toAddress === NULL_ADDRESS
+
+    return {
+      key: transfer.tx_hash!,
```

Comment:
This is not a reliable key. Transactions emit multiple events. I think that's why pagination is misbehaving:
- https://0d99ad56.oasis-explorer.pages.dev/mainnet/emerald/address/0xf02b3e437304892105992512539F769423a515Cb/token-transfers
- see 10 transfers
- go to page 2, go to page 1
- see 26 transfers
- go to page 2, go to page 1
- 56

Reply:
> This is not a reliable key. Transactions emit multiple events. 

Right. Moved to a composite index, which also involves index number within the loaded list of events. After this, I can't reproduce the issue any more.